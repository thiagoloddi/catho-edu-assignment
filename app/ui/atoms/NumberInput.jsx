import React from 'react';
import injectSheet from 'react-jss';
import propTypes from 'prop-types';

const NumberInput = ({ classes, value, id, onChange, onClick, onBlur }) => {
  return (
    <div className={classes.container}>
      <img onClick={onClick} id="minus"className={classes.img} src="/assets/minus.svg"/>
      <input id={id} value={value} onChange={onChange} onBlur={onBlur} className={classes.input}/>
      <img onClick={onClick} id="plus"className={classes.img} src="/assets/plus.svg"/>
    </div>
  )
}

const style = {
  container: {
    height: '100',
    '& > *': {
      display: 'inline-block',
      verticalAlign: 'middle'
    }
  },
  img: {
    height: '15px',
    width: '15px',
    cursor: 'pointer'
  },
  input: {
    width: '30px',
    margin: '0 10px',
    textAlign: 'center',
    outline: 'none'
  }
}

NumberInput.propTypes = {
	value: propTypes.number.isRequired,
	id: propTypes.string.isRequired,
	onChange: propTypes.func.isRequired,
	onClick: propTypes.func.isRequired,
	onBlur: propTypes.func
};

export default injectSheet(style)(NumberInput);