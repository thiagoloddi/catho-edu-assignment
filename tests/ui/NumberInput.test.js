import React from 'react';
import renderer from 'react-test-renderer';
import NumberInput from '../../app/ui/atoms/NumberInput';

const func = () => {};

it('renders correctly', () => {
  const tree = renderer
    .create(<NumberInput value={0} id={'1'} onChange={func} onClick={func} onBlur={func} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});