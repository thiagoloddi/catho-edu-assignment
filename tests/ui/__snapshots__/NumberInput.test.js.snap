// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="NumberInput-container-0-1-1"
>
  <img
    className="NumberInput-img-0-1-2"
    id="minus"
    onClick={[Function]}
    src="/assets/minus.svg"
  />
  <input
    className="NumberInput-input-0-1-3"
    id="1"
    onBlur={[Function]}
    onChange={[Function]}
    value={0}
  />
  <img
    className="NumberInput-img-0-1-2"
    id="plus"
    onClick={[Function]}
    src="/assets/plus.svg"
  />
</div>
`;
